lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.html

```diff
@@ -48,26 +94,50 @@
+            <mat-expansion-panel-header>
+                <adf-content-metadata-header [title]="'CATEGORIES_MANAGEMENT.CATEGORIES_TITLE'" [expanded]="isCategoriesPanelExpanded">
+                    <div class="adf-categories-button">
+                        <button *ngIf="canEditCategories"
+                                mat-icon-button
+                                (click)="onToggleCategoriesEdit($event)"
+                                [attr.title]="'CORE.METADATA.ACTIONS.EDIT' | translate"
+                                [attr.aria-label]="'CORE.METADATA.ACCESSIBILITY.EDIT' | translate"
+                                data-automation-id="meta-data-categories-edit"
+                                class="adf-edit-icon-buttons">
+                            <mat-icon>mode_edit</mat-icon>
+                        </button>
+                    </div>
+                    <div *ngIf="isEditingCategories" class="adf-metadata-action-buttons">
+                        <button mat-icon-button
+                                [attr.title]="'CORE.METADATA.ACTIONS.CANCEL' | translate"
+                                (click)="onCancelCategoriesEdit($event)"
+                                data-automation-id="reset-metadata"
+                                class="adf-metadata-action-buttons-clear">
+                            <mat-icon>clear</mat-icon>
+                        </button>
+                        <button mat-icon-button
+                                [attr.title]="'CORE.METADATA.ACTIONS.SAVE' | translate"
+                                (click)="onSaveCategoriesChanges($event)"
+                                color="primary"
+                                data-automation-id="save-categories-metadata"
+                                [disabled]="!hasMetadataChanged">
+                            <mat-icon>check</mat-icon>
+                        </button>
+                    </div>
+                </adf-content-metadata-header>
+            </mat-expansion-panel-header>
+            <div *ngIf="!isEditingCategories">
                 <p *ngFor="let category of categories" class="adf-metadata-categories">{{ category.name }}</p>
-            </mat-expansion-panel>
-            <div *ngIf="editable"
-                class="adf-metadata-categories-header">
-                <div class="adf-metadata-categories-title">
-                    <p>{{ 'CATEGORIES_MANAGEMENT.CATEGORIES_TITLE' | translate }}</p>
-                    <button
-                        mat-icon-button
-                        [attr.title]="'CATEGORIES_MANAGEMENT.ASSIGN_CATEGORIES' | translate"
-                        [hidden]="categoryControlVisible || saving"
-                        (click)="categoryControlVisible = true">
-                        <mat-icon>add</mat-icon>
-                    </button>
-                </div>
+            </div>
+            <div *ngIf="showEmptyCategoryMessage()" class="adf-metadata-no-item-added">
```

Comment:
We raised some time ago jira to avoid calling functions in html template except for events. Can we replace that function's usage with some variables?

Reply:
@DenysVuika same here, should we address that?